test/extensions/filters/http/transformation/inja_transformer_test.cc

```diff
@@ -565,6 +601,30 @@ TEST(InjaTransformer, ParseBodyListUsingContext) {
   EXPECT_EQ(body.toString(), "321");
 }
 
+TEST(InjaTransformer, ParseFromClusterMetadata) {
+  Http::TestHeaderMapImpl headers{{":method", "GET"}, {":path", "/foo"}};
+  TransformationTemplate transformation;
+  transformation.mutable_body()->set_text("{{clusterMetadata(\"key\")}}");
+
+  InjaTransformer transformer(transformation);
+
+  NiceMock<Http::MockStreamDecoderFilterCallbacks> callbacks;
+  
+  Upstream::MockClusterInfo* cluster_info = new NiceMock<Upstream::MockClusterInfo>();
+  Upstream::ClusterInfoConstSharedPtr cluster_info_ptr{cluster_info};
+
+  envoy::config::core::v3::Metadata meta;
+  meta.mutable_filter_metadata()->insert({SoloHttpFilterNames::get().Transformation, MessageUtil::keyValueStruct("key", "val")});
+  ON_CALL(*cluster_info, metadata()).WillByDefault(testing::ReturnRefOfCopy(meta));
```

Comment:
```suggestion
  ON_CALL(*callbacks.cluster_info_, metadata()).WillByDefault(testing::ReturnRefOfCopy(meta));
```
and then you can get rid of the cluster_info_ptr and the EXPECT_CALL below